ribs/emitters/_gaussian_emitter.py

```diff
@@ -99,29 +101,35 @@ def batch_size(self):
     def ask(self):
         """Creates solutions by adding Gaussian noise to elites in the archive.
 
-        If the archive is empty, solutions are drawn from a (diagonal) Gaussian
-        distribution centered at ``self.x0`` with standard deviation
-        ``self.sigma0``. Otherwise, each solution is drawn from a distribution
-        centered at a randomly chosen elite with standard deviation
-        ``self.sigma``.
+        If the archive is empty and ``self._initial_solutions`` is set, we
+        return ``self._initial_solutions``. If ``self._initial_solutions`` is
+        not set, we draw from Gaussian distribution centered at ``self.x0``
+        with standard deviation ``self.sigma``. Otherwise, each solution is
+        drawn from a distribution centered at a randomly chosen elite with
+        standard deviation ``self.sigma``.
 
         Returns:
-            ``(batch_size, solution_dim)`` array -- contains ``batch_size`` new
-            solutions to evaluate.
+            If the archive is not empty, ``(batch_size, solution_dim)`` array
+            -- contains ``batch_size`` new solutions to evaluate. If the
+            archive is empty, we return ``self._initial_solutions``, which
+            might not have ``batch_size`` solutions.
         """
-        if self.archive.empty:
-            sigma = self._sigma0
-            parents = np.expand_dims(self._x0, axis=0)
-        else:
-            sigma = self._sigma
-            parents = self.archive.sample_elites(
-                self._batch_size).solution_batch
-
         noise = self._rng.normal(
-            scale=sigma,
+            scale=self._sigma,
             size=(self._batch_size, self.solution_dim),
         ).astype(self.archive.dtype)
 
+        if self.archive.empty:
+            if self._initial_solutions is None:
+                return np.clip(
+                    np.expand_dims(self.x0, axis=0) + noise,
+                    self.lower_bounds,
+                    self.upper_bounds,
+                )
+            return np.clip(self._initial_solutions, self.lower_bounds,
```

Comment:
This is wasting effort because `noise` is generated even if we just return the initial solutions. Can you fix this?